### PR TITLE
start of docs

### PR DESCRIPTION
**Pull request by darcys22**

### CODE REVIEW

**Comment by darcys22** [comment]
Still working through this, thoughts are that there should be 2 api pages. One for the `C` headers and another for the `CPP` headers

**Comment by darcys22** [comment]
Still left to do for C headers - `user_groups.h`

Should investigate a way to get structs into the docs aswell rather than just functions

File: docs/api/make-docs.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+else
+    echo "Can't determine your node_modules path; set NPM_PACKAGES or NODE_PATH appropriately" >&2
+    exit 1
+fi
+
+cp $npm_dir/docsify/node_modules/prismjs/components/prism-{json,python}.min.js api/vendor
+
+./api-to-markdown.py "$@"
+
+perl -ni.bak -e '
```
Comment (majestrate) [inline docs/api/make-docs.sh]: may be nicer to use jinja2 templates to do this step, that would also let you do it all in that python script

Comment (majestrate) [reply docs/api/make-docs.sh]: (I may volunteer myself this task if it doesn't conflict too hard)

**Pull request merged**